proposals/3392/dataset_partitioning.md

```diff
@@ -0,0 +1,550 @@
+   "PartitionSubsetCondition": {
+      "allOf": [
+         {
+            "$ref": "#/$defs/BaseSubsetCondition"
+         },
+         {
+            "type": "object",
+            "properties": {
+               "partitions": {
+                  "type": "array",
+                  "items": {
+                     "types": "object",
+                     "properties": {
+                        "identifier": {
+                           "type": "string",
+                           "description": "Optionally provided identifier of the partition specified"
+                        },
+                        "type":  {
+                           "const": "partition"
+                        },
+                        "dimensions": {
+                           "type": "object",
+                           "additionalProperties": true
+                        }
+                     }
+                  }
+               }
+            }
+         }
+      ]
+   }
+}
+```
+
+Additional to logical subset definition, there is a need to describe physical locations. 
+This can be:
+```json
+{
+   "LocationSubsetCondition": {
+      "allOf": [
+         {
+            "$ref": "#/$defs/BaseSubsetCondition"
+         },
+         {
+            "type": "object",
+            "properties": {
+               "type":  {
+                  "const": "location"
+               },
+               "locations": {
+                    "type": "array",
+                    "items": {
+                      "type": "string"
+                    }
+               }
+            }
+         }
+      ]
+   }
+}
+```
+
+A Spark job may read dataset from a location with a wildcard, like `/some/path/a=*/b=1`. In this case
+we assume, subset definition can be resolved from physical plan when wildcard gets resolved into a 
+list of files.
+
+In some cases, like  SQL queries, a subset definition can be too complex to be represented in a structured
+way. However, this still makes sense to include a description of it.
+
+```json
+{
+   "DescriptionSubsetCondition": {
+      "allOf": [
+         {
+            "$ref": "#/$defs/BaseSubsetCondition"
+         },
+         {
+            "type": "object",
+            "properties": {
+               "type":  {
+                 "const": "description"
+               },
+               "description": "String"
+            }
+         }
+      ]
+   }
+}
+```
+
+In other words, we accept OpenLineage will not fully understand SQL syntax, like the syntax below:
+```sql
+SELECT *
+FROM Employees
+WHERE EXISTS (SELECT * FROM EmployeeReviews er  WHERE er.EmployeeID = Employees.EmployeeID AND er.Rating > 4);
+```
+However, even in this case, OpenLineage should be able to provide informative subset definition:
+```json
+{
+   "subset": {
+      "type": "description",
+      "description": "EXISTS (SELECT * FROM EmployeeReviews er  WHERE er.EmployeeID = Employees.EmployeeID AND er.Rating > 4)"
+   }
+}
+```
+
+### Subset dataset facet
```

Comment:
Q: since the notion of subset is more generic than that of partition (be it logical or physical), do you see the present proposals covering also bucketing?

Reply:
Including bucketing within the proposal makes sense only if we know this is achievable for some OL producers, like for example Spark. I am not sure if a list of updated/read buckets can be extracted with spark connector. And I would like to avoid verifying this when working on the proposal. 

However, I am OK to add a comment that hierarchy model can be extended with `bucket` subset type in future.